server/jobWorkerServer.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/P-A-R-U-S/Go-Job-Worker-Service/pkg/jobWorker"
+	"github.com/P-A-R-U-S/Go-Job-Worker-Service/pkg/proto"
+	"github.com/P-A-R-U-S/Go-Job-Worker-Service/pkg/tls"
+	"google.golang.org/grpc"
+	"io"
+	"sync"
+)
+
+var (
+	ErrJobNotFound   = errors.New("job not found")
+	ErrNotAuthorized = errors.New("user is not authorized to access job")
+)
+
+type userJob struct {
+	user string
+	job  *jobWorker.Job
+}
+
+type JobWorkerServer struct {
+	userJobs map[string]userJob
+	mutex    sync.Mutex
+}
+
+func NewJobWorkerServer() *JobWorkerServer {
+	return &JobWorkerServer{
+		userJobs: map[string]userJob{},
+	}
+}
+
+// Start creates a new job for the user and starts the job.
+func (s *JobWorkerServer) Start(ctx context.Context, request *proto.JobCreateRequest) (*proto.JobResponse, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	user, err := tls.GetUserFromContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user from certificate: %w", err)
+	}
+
+	config := jobWorker.JobConfig{
+		CPU:              request.CPU,
+		IOBytesPerSecond: request.IoBytesPerSecond,
+		MemBytes:         request.MemBytes,
+		Command:          request.GetCommand(),
+		Arguments:        request.GetArgs(),
+	}
+
+	newJob := jobWorker.NewJob(&config)
+
+	s.userJobs[newJob.UUID.String()] = userJob{
+		user: user,
+		job:  newJob,
+	}
+
+	if err = newJob.Start(); err != nil {
+		return &proto.JobResponse{Id: newJob.UUID.String()}, fmt.Errorf("error starting job: %w", err)
```

Comment:
In base case if server returns empty response or nil - error would be same 

![Screenshot 2024-09-09 at 1 39 55 PM](https://github.com/user-attachments/assets/09a93b19-33f8-4764-87cf-d317e385a4c4)

**Note:** But I would return response with JobID and error and let client handle it, rather than not return empty response. Because we would use UUID in real production and client/customer always able to provide use some UUID and let us investigate why job wasn't started, failed or etc.